src/main/java/org/broadinstitute/hellbender/tools/spark/pathseq/PSUtils.java

```diff
@@ -101,6 +104,10 @@ public static int getMatchesLessDeletions(final Cigar cigar, final int numMismat
                 numDeletions += e.getLength();
             }
         }
+        if (numMatches < 0)  {
+            logger.warn("Invalid arguments passed to getMatchesLessDeletions(): numMismatches was greater than the number of matches/mismatches in the cigar. Returning 0.");
```

Comment:
This is an infrequent occurrence, right? Otherwise I would place a max count of warning messages. Since it all be done inside a (infrequent-run) if body then you don't need to worry a bout using slower synchronization. Only if you notice that the amount of warning messages is large.

Reply:
It is relatively infrequent (a small proportion of reads). I will leave it as is.